### PR TITLE
use the same base class for Label and OneHotLabel

### PR DESCRIPTION
**Pull request by pmeier**
Follow-up to https://github.com/pytorch/vision/pull/6420#issuecomment-1225374947. TL;DR: having a common base class for `Label` and `OneHotLabel` let's us reuse some common logic without risking one being mistaken for the other in the transforms.

### CODE REVIEW

File: torchvision/prototype/features/_label.py [hunk 1]
```diff
@@ -1,57 +1,60 @@
...
+from typing import Any, Optional, Sequence, Type, TypeVar, Union
 
 import torch
 from torch.utils._pytree import tree_map
 
 from ._feature import _Feature
 
 
-class Label(_Feature):
+F = TypeVar("F", bound="_Feature")
```
Comment (datumbox) [inline torchvision/prototype/features/_label.py:11]: Did you mean the following?
```suggestion
F = TypeVar("F", bound="_LabelBase")
```

If yes should it be `L` instead of `F`?

**Review by datumbox — APPROVED** [review]
LGTM, just one question:
